Dockerfile

```
@@ -1,16 +1,35 @@
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.5 AS builder
+
+RUN microdnf install -y tar gzip make which
+
+# install go 1.17.8
+RUN curl -O -J https://dl.google.com/go/go1.17.8.linux-amd64.tar.gz
+RUN tar -C /usr/local -xzf go1.17.8.linux-amd64.tar.gz
+RUN ln -s /usr/local/go/bin/go /usr/local/bin/go
+
+WORKDIR /workspace
+
+COPY go.mod ./
+COPY go.sum ./
+RUN go mod download
```

It is recommended to use `go mod tidy`. This downloads dependencies and removes deps which are not necessary.

```suggestion
RUN go mod tidy
```

---

Not an expert on Golang build tools and procedures here, but can you quote a source for this claim?
My sources are:
* (the official Docker manual)[https://docs.docker.com/language/golang/build-images/]
* (Medium article)[https://petomalina.medium.com/using-go-mod-download-to-speed-up-golang-docker-builds-707591336888]
* (Codefresh article)[https://codefresh.io/docs/docs/learn-by-example/golang/golang-hello-world/]
among others.

The worry I am having regarding "go mod tidy" is that it can, as a side-effect, modify `go.sum`, which would render a given repo as dirty. If, as we do in StackRox, image tags are derived from Git revision information, this would produce undesired behaviour, I guess.

---

Yeah, a lot of docs point to `go mod download` 

`go mod tidy` ensures that the dependencies match the source. If `go.sum` changed after running `go mod tidy` the build should ideally fail because dependencies are out of date.
In stackrox builds often failed because of `go mod download` because deps were out of sync.

Compare docs between [tidy](https://go.dev/ref/mod#go-mod-tidy) and [download](https://go.dev/ref/mod#go-mod-download).

I can't find the github discussion anymore though on this topic...